src/hddsupport.c

```diff
@@ -93,16 +93,63 @@ static int hddCheckHDProKit(void)
 #define PFS_ZONE_SIZE 8192
 #define PFS_FRAGMENT 0x00000000
 
-static int CreateOPLPartition(const char *oplPart, const char *mountpoint)
+static int hddFindOPLPartition(void)
+{
+    static config_set_t *config;
+    char name[64];
+    int fd, ret = 0;
+
+    fileXioUmount(hddPrefix);
+
+    ret = fileXioMount(hddPrefix, "hdd0:__common", FIO_MT_RDWR);
+    if (ret == 0) {
+        fd = open("pfs0:conf_hdd.cfg", O_RDONLY);
+        if (fd >= 0) {
+            config = configAlloc(0, NULL, "pfs0:conf_hdd.cfg");
+            configRead(config);
+
+            configGetStrCopy(config, "hdd_partition", name, sizeof(name));
+            snprintf(oplPart, sizeof(oplPart), "hdd0:%s", name);
+
+            configFree(config);
+            close(fd);
+
+            if (!strcmp(name, "__common"))
+                return 0;
```

Comment:
just maybe it will be safer to unmount partition also before this return

Reply:
or just mobe `fileXioUmount(hddPrefix);` before `if (!strcmp(name, "__common"))`. You will unmount the partition regardless of next steps.

Reply:
We can unmount but the idea was to just return and not remount any other partition because at this point __common is mounted and the cfg says to use __common as OPL part

Reply:
I understand - you don't want to spend time unmounting/mounting pfs0 if we will use the same partition (`__common`). but the code becomes less readable due to that fact. Anyway, I advise mounting `__common` in `pfs1` and not unmount it at all.